Controller/CustomItem/SaveController.php

```diff
@@ -136,18 +136,24 @@ public function saveAction(int $objectId, ?int $itemId = null): Response
         $form->handleRequest($request);
 
         if ($form->isValid()) {
-            $this->customItemModel->save($customItem);
+            $itemCount = $this->customItemModel->getRepository()->count(['customObject' => $this->customObjectModel->fetchEntity($objectId)->getId()]);
+            $customItem = $this->customItemModel->save($customItem);
 
             if ($customItem->getChildCustomItem()) {
                 $customItem->getChildCustomItem()->generateNameForChildObject('contact', $contactId, $customItem);
-                $this->customItemModel->save($customItem->getChildCustomItem());
+                $customItem = $this->customItemModel->save($customItem->getChildCustomItem());
+            }
+
+            if(!$itemId && $itemCount === $this->customItemModel->getRepository()->count(['customObject' => $this->customObjectModel->fetchEntity($objectId)->getId()])){
```

Comment:
This is not going to work in all cases. Imagine there is import happening when saving a custom item. The worker can store a couple of items between fetching the first count and second. It will show the "merged" message even it is not true.

Also, counts are very expensive queries, so it may happen that the save operation will never finish just because we are creating 2 expensive queries to figure out whether the entity was updated or inserted.

I suggest to create new interface called `Upsertable` and define new methods in it:
```php
public function setWasUpdated(): void;
public function setWasInserted(): void;
public function wasUpdated(): bool;
public function wasInserted(): bool;
```
Feel free to create trait to implement these methods so each Upsertable entity does not have to do it by itself.

Then in https://github.com/acquia/mc-cs/blob/staging/app/bundles/CoreBundle/Entity/CommonRepository.php#L863 you'd check whether the entity implements the interface and if so, set if the entity was updated or inserted based on this:

https://www.php.net/manual/en/mysqli.affected-rows.php

Reply:
Thanks for the these reviews, it all makes so much more sense sense than what I had implemented. I have made the changes, please take a look.